wrappingGsdk/ProcessSample/wrapper/Program.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using Microsoft.Playfab.Gaming.GSDK.CSharp;
+using System.IO;
+using System.Reflection;
+using System.Linq;
+
+namespace wrapper
+{
+    class Program
+    {
+        private static Process gameProcess;
+        private static IDictionary<string, string> activeConfig;
+        private const string portName = "RealPort";
+        private const string commandExe = "cmd";
+        private static string _listeningPort = null;
+
+
+        static void Main(string[] args)
+        {
+            if (args.Length <= 1 || args[0] != "-g")
+            {
+                Console.WriteLine("Usage: wrapper.exe -g fakegame.exe args...");
+                return;
+            }
+
+            string gameserverExe = args[1];
+
+            // check here for the full guide on integrating with the GSDK 
+            // https://docs.microsoft.com/en-us/gaming/playfab/features/multiplayer/servers/integrating-game-servers-with-gsdk
+
+            LogMessage("Wrapper sample for Azure PlayFab Multiplayer Servers");
+            
+            LogMessage("Attempting to register GSDK callbacks");
+            RegisterGSDKCallbacksAndStartGSDK();
+            LogMessage("GSDK callback registration completed");
+            
+            LogMessage("Attempting to start game process");
+            InitiateAndWaitForGameProcess(gameserverExe, args.Skip(2));
+            LogMessage("Game process has exited");
+        }
+
+        // starts main game process and wait for it to complete
+        public static void InitiateAndWaitForGameProcess(string gameserverExe, IEnumerable<string> args)
+        {
+            activeConfig = GameserverSDK.getConfigSettings();
+
+            // here we're starting the script that initiates the game process
+            if (activeConfig.TryGetValue(portName, out string listeningPortString))
+            {
+                GameserverSDK.LogMessage($"{portName}:{listeningPortString} was found in GSDK Confg Settings.");
+                _listeningPort = listeningPortString;
+            }
+            else
+            {
+                LogMessage($"Cannot find {portName} in GSDK Config Settings. Please make sure the MockAgent is running " +
+                             $"and that the MultiplayerSettings.json file includes correct {portName} as a GamePort Name.");
+                return;
+            }
+
+            Process processPortInUse = GetProcessPortInUse();
+
+            // port in use 
+            if (processPortInUse != null)
+            {
+                string fakeGameProcessName = gameserverExe.Split(".")[0];
+                if (processPortInUse.ProcessName == fakeGameProcessName)
+                {
+                    Console.WriteLine($"Port:{_listeningPort} is already in use.");
```

Comment:
why would this happen?

Reply:
game server is not terminated gracefully so still running on the port use specified. ( node port )

Reply:
that's the case when running on the LocalMultiplayerAgent I guess.... instead of killing it, maybe we can output a message or something? On production something might happen and the server have active players, we wouldn't want to kick them out

Reply:
Gotcha I will just show the logs for this case.

Reply:
Should I terminate the game if port number grabbed from GSDK is already in use?

Reply:
I'd rather just show an error and then terminate the wrapper process. If this ever happens, something is wrong with the system.